### DISCUSSION TITLE
How would you go about adding error checking?

### DISCUSSION THREAD

**Discussion by podfeet** [Honatas/bootstrap-4-autocomplete#26] [Q&A]
I'm using this library to great success, but I want to do error checking on whether the user has actually selected anything from the dropdown. Want an error thrown under three conditions:

- They type something and don't get a match
- They type something which does have a match, but they don't select anything
- They type something and tab away to the next search box (which might be the same as the first two)

I think there's a handler on keyup, which I think is triggered when the person selects from the list or uses arrow keys and hits enter, but I'm not sure how to access that to throw an error if they _don't_ keyup. I'm hoping error checking is already written like maybe in the Bootstrap dropdown, but I haven't found it there yet.

Here's how I'm using @Honatas's library:

https://nosillacast.github.io/time-shifter-clock/

**Comment by Honatas** [comment]
Hi @NosillaCast !

I believe you can create an implementation to check if the value of the textfield is contained inside your source data at the textfield's **[blur](https://www.w3schools.com/jsref/met_html_blur.asp)** event. You could also check if the callback **onSelectItem** has been fired just by implementing it, for it confirms a selection - just set another variable or hidden field whenever the method is fired and check it instead of checking the text field.

These validations are very specific to each use case, that's why I didn't add any.

**Reply by podfeet** [reply]
Thanks @Honatas  - I didn't put this under issues because I suspected it was specific to use case. 

I thought about checking the text field, but I don't think I want to be that helpful to my users! I want to merely check to see if they've selected something so checking to see if **onSelectItem** has fired is a great idea.

I was starting to go down the **keyup** path but I was struggling with when to check for that.  I'll give your idea a try.
